clang/test/3C/placements.c

```diff
@@ -11,11 +11,11 @@
 // expected-no-diagnostics
 void what(const char *s, int q);
 //CHECK_NOALL: void what(const char *s : itype(_Ptr<const char>), int q);
-//CHECK_ALL: void what(_Array_ptr<const char> s : count(q), int q);
+//CHECK_ALL: void what(_Array_ptr<const char> s : count(5 + 1), int q);
```

Comment:
If understand what's happening here,  the old bound was `q` because one of our bounds heuristics is that if an array variable is followed by an integer then the integer is the bound for the array. The new bounds are constant because bound inference is able to assign a bound to `s` before considering heuristics. Since it already has a bound, heuristics aren't used.

I think the new bound is probably better than the old.

Reply:
You are right.